app/models/activity.rb

```diff
@@ -0,0 +1,58 @@
+class Activity < ActiveRecord::Base
+  before_create { |activity| activity.event = generate_key(activity.object, activity.event) }
+
+  belongs_to :object, polymorphic: true
+  belongs_to :actor, inverse_of: :activities, class_name: User.name
+  has_many :course_notifications, inverse_of: :activity, dependent: :destroy
+  has_many :user_notifications, inverse_of: :activity, dependent: :destroy
+
+  USER_NOTIFICATION_TYPES = [:popup, :email]
+  COURSE_NOTIFICATION_TYPES = [:feed]
+
+  # Create notifications based on an activity.
+  #
+  # @param [User, Course] to the user who will receive notifications or the course which will show
+  #   activity feeds.
+  # @param [Array<Symbol>, Symbol] types the formats of the notifications. If 'to' is a user 'types'
+  #   can be :popup, :email or both. If 'to' is a course 'types' can only be :feed.
+  def notify(to, types: nil)
```

Comment:
now that your notify method chains, you did not allow overloading the type/types keyword argument. `...notify(user, types: :popup)` looks weird, right?